pageserver/src/tenant/filename.rs

```diff
@@ -172,6 +172,118 @@ impl fmt::Display for ImageFileName {
         )
     }
 }
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub enum LayerFileName {
+    Image(ImageFileName),
+    Delta(DeltaFileName),
+    #[cfg(test)]
+    Test(String),
+}
+
+impl LayerFileName {
+    pub fn file_name(&self) -> String {
+        match self {
+            LayerFileName::Image(fname) => format!("{fname}"),
+            LayerFileName::Delta(fname) => format!("{fname}"),
+            #[cfg(test)]
+            LayerFileName::Test(fname) => fname.to_string(),
+        }
+    }
+    #[cfg(test)]
+    pub(crate) fn new_test(name: &str) -> LayerFileName {
+        LayerFileName::Test(name.to_owned())
+    }
+}
+
+impl From<ImageFileName> for LayerFileName {
+    fn from(fname: ImageFileName) -> Self {
+        LayerFileName::Image(fname)
+    }
+}
+impl From<DeltaFileName> for LayerFileName {
+    fn from(fname: DeltaFileName) -> Self {
+        LayerFileName::Delta(fname)
+    }
+}
+
+// include a `/` in the name as an additional layer of robustness
+// because `/` chars are not allowed in UNIX paths
+#[cfg(test)]
+pub const LAYER_FILE_NAME_TEST_PREFIX: &str = "LAYER_FILE_NAME::test/";
+
+impl<'a> TryFrom<&'a str> for LayerFileName {
```

Comment:
Style NIT: Rust stdlib provides `FromStr` as an alternative https://doc.rust-lang.org/std/str/trait.FromStr.html
that enables a `parse` method on every &str, hence no later trait imports would needed to use the conversion.

`parse` name is also looks more applicable due to `parse_str` invocations inside.

Reply:
Switched to `FromStr`